provider/datagen/src/transform/cldr/percent/mod.rs

```diff
@@ -0,0 +1,92 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::collections::HashSet;
+
+use crate::provider::IterableDataProviderInternal;
+use crate::transform::cldr::cldr_serde;
+
+use icu_dimension::provider::*;
+use icu_provider::prelude::*;
+use icu_provider::DataProvider;
+use tinystr::tinystr;
+
+impl DataProvider<PercentEssentialsV1Marker> for crate::DatagenProvider {
+    fn load(&self, req: DataRequest) -> Result<DataResponse<PercentEssentialsV1Marker>, DataError> {
+        self.check_req::<PercentEssentialsV1Marker>(req)?;
+        let langid = req.locale.get_langid();
+
+        let numbers_resource: &cldr_serde::numbers::Resource = self
+            .cldr()?
+            .numbers()
+            .read_and_parse(&langid, "numbers.json")?;
+
+        let result = extract_percent_essentials(numbers_resource);
+
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(result?)),
+        })
+    }
+}
+
+impl IterableDataProviderInternal<PercentEssentialsV1Marker> for crate::DatagenProvider {
+    fn supported_locales_impl(&self) -> Result<HashSet<DataLocale>, DataError> {
+        Ok(self
+            .cldr()?
+            .numbers()
+            .list_langs()?
+            .map(DataLocale::from)
+            .collect())
+    }
+}
+
+fn extract_percent_essentials<'data>(
+    numbers_resource: &cldr_serde::numbers::Resource,
+) -> Result<PercentEssentialsV1<'data>, DataError> {
+    let percent_formats = &&numbers_resource
+        .main
+        .value
+        .numbers
+        .numsys_data
+        .percent_patterns
+        .get(&tinystr!(8, "latn"))
+        .ok_or_else(|| DataError::custom("Could not find the standard pattern"))?;
+
+    let standard = &percent_formats.standard;
+
```

Comment:
Somewhere in here you'll want to do the following:

1. Remove the number placeholder, like `#,##0`, from the string, and keep track of its index in the string (will often but not necesarilly be zero)
2. Replace the `%` character with the localized percent symbol, which you should be able to pull also from the CLDR JSON data file

Reply:
Updated!